lib/diagnostic.js

```diff
@@ -14,12 +14,30 @@
 'use strict';
 
 const fs = require('fs'); // jshint ignore: line
+const stdin = '/dev/stdin';
 
-const sumLines = (filename, callback) => {
-  /* your response here */
-  callback();
+let inFile = process.argv[2] === '-' ? stdin : process.argv[2];
+
+const sumLines = (array /*filename, callback*/) => {
+
+  let result;
+  for (let i = 0; i < array; i++) {
+    result += array[i];
```

Comment:
Remember that Javascript does not care about datatype when it comes to `+=`. This operation will work with both strings and numbers. With strings, it will simply append them.